### PR TITLE
Add support for cy.rails() to evaluate code in Rails test server

### PR DESCRIPTION
**Pull request by valscion**
This PR adds support for running Ruby code inside Cypress tests.

For example, this works:

> ```js
> cy.rails(`
>   "Hello world!"
> `).then(result => {
>   expect(result).to.equal('Hello world!')
> })
> ```
> 
> This evaluates the `"Hello world!"` in Ruby context, returning the JSON string "Hello world!" to Cypress JavaScript environment and then Cypress code asserts that the string is indeed what we expected.
> 
> video-game-database-ruby

We also have access to the Rails named route helpers via `Routes` module and that can be used like this:

> 
> ```js
> cy.rails(`Routes.new_user_registration_path`).then(path => cy.visit(path))
> ```
> 
> video-game-database-ruby

If the result that we try to evaluate can't be converted to JSON, then `null` will be returned to Cypress code.

> ```js
> cy.rails(`
>   # This will result in an ActiveRecord::Relation object
>   # and that can't be serialized to JSON
>   Game.where(name: "query")
> `).then(result => {
>   expect(result).not.to.equal(null)
> })
> ```
> 
> video-game-database-ruby

If there's a Ruby syntax error the code tries to show the error in a nice way:

> ```js
> cy.rails(`
>   "Oh no I forgot a closing quote
> `)
> ```
> 
> video-game-database-ruby

If there's a Ruby some other error, we still try to print a nice error:

> ```js
> cy.rails(`
>   Game.find(123)
> `)
> ```
> 
> video-game-database-ruby
> 
>

### CODE REVIEW

File: cypress/support/rails.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+Cypress.Commands.add('rails', function railsTask(code) {
```
Comment (valscion) [inline cypress/support/rails.js:1]: You don't need to understand anything about this file, so feel free to ignore it :relaxed:

File: lib/venuu-cypress/code_formatter.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
```
Comment (valscion) [inline lib/venuu-cypress/code_formatter.rb:1]: You don't need to understand anything about this file, so feel free to ignore it :relaxed:

File: lib/venuu-cypress/eval_context.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
```
Comment (valscion) [inline lib/venuu-cypress/eval_context.rb:1]: You don't need to understand anything about this file, so feel free to ignore it :relaxed:

File: lib/venuu-cypress/middleware.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
```
Comment (valscion) [inline lib/venuu-cypress/middleware.rb:1]: You don't need to understand anything about this file, so feel free to ignore it :relaxed:

**Comment by valscion** [comment]
I'll merge these two pull requests (this and #33) now so that we get working `cy.rails()` commands here. The PR descriptions can still be checked even after the PRs have been merged to better understand how we can use `cy.rails()` :relaxed: @venuu/ob

**Comment by jraunio** [comment]
👍

**Comment by JonasThers** [comment]
Makes sense 👍

**Pull request merged**